### PR TITLE
Add Jenkinsfile and initial skeleton for the EdgeX Release

### PR DESCRIPTION
**Pull request by lranjbar**
Signed-off-by: Lisa Rashidi-Ranjbar <lisa.a.rashidi-ranjbar@intel.com>

### CODE REVIEW

File: release/sample-service.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+version: 'v1.2.0'
+releaseName: 'geneva'
+releaseStream: 'master'
+repo: 'https://github.com/edgexfoundry/sample-service.git'
+gitTag: true
+gitTagDestination: 'https://github.com/edgexfoundry/sample-service.git'
+dockerImages: true
+dockerSource:
+  - 'https://nexus3.edgexfoundry.org/..'
+  - 'https://nexus3.edgexfoundry.org/..'
```
Comment (mkbhanda) [inline release/sample-service.yaml]: why do we have two instances of this dockerSource?

File: release/test-service.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+name: 'test-service'
+version: 'v1.2.0'
+releaseName: 'geneva'
+releaseStream: 'master'
+repo: 'https://github.com/edgexfoundry/sample-service.git'
+gitTag: false
+gitTagDestination: 'https://github.com/edgexfoundry/sample-service.git'
+dockerImages: true
+dockerSource:
+  - 'https://nexus3.edgexfoundry.org/..'
```
Comment (mkbhanda) [inline release/test-service.yaml]: ditto from above

File: release/sample-service.yaml (hunk 1 above)
Comment (lranjbar) [reply release/sample-service.yaml]: It is meant to be a list of docker images to release for the repository. All of the repositories in EdgeX have more than one docker image to release.

**Comment by lranjbar** [comment]
> Some few questions:
> 
> 1. how will specific releases be tracked, are we going to have different branches representing releases?
> 2. the yaml definition appears to be self-explanatory for the most part, however I'm not sure what the repeated entries mean for some of them, for example `dockerSource`
> 
> Looks really cool, looking forward to your demo.

1. Different named/version releases will be tracked using the same branch. This design sets that for every repository we will have one YAML file to describe the release. When we update the YAML files for the new release it will create an audit trail in the git commit history for that file/repository.
2. The repeated entries in dockerSource are because we will have more than one docker image to release for each repository. The YAML file is a best guess on what information we will need to call the global functions. As we add functionality to the global functions we will iterate on the YAML file.

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        timestamps()
+        timeout(360)
+    }
+    stages {   
+        stage('Lint YAML files') {
+            steps {
+                script {
+                    def yamllintImage = "nexus3.edgexfoundry.org:10003/edgex-devops/yamllint:latest"
+
+                    docker.image(yamllintImage).inside("-v ${env.WORKSPACE}:/ws -w /ws --entrypoint=''") {
```
Comment (ernestojeda) [inline Jenkinsfile]: You shouldn't have to bind mount the workspace to /ws. By default when you use docker.image.inside it automatically mount the workspace in the image.

File: Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+                }
+            }
+        }
+        stage('Prepare Release YAML') {
+            when { 
+                expression { env.GIT_BRANCH == 'release'} 
+            }
+            steps {
+                script {
+                    def steps = [:]
```
Comment (ernestojeda) [inline Jenkinsfile]: Why is this here twice? Same as line 21

**Comment by lranjbar** [comment]
recheck
